pieces.h

```diff
@@ -1,106 +1,67 @@
-#ifndef Piece_Class_H
-#define Piece_Class_H
-
-#include <algorithm>
-
-using namespace std;
+#pragma once
 
+#include <vector>
+#include "constants.h"
 
 class Piece {
 public:
-    // playfield is 10 wide 40 tall.
-    // cell called using y * 10 + x
-    static bool* board[400];
-};
-
-class Square : public Piece {
-    int pos;
+    int color;
+    // position of pieces relative to position of piece given in board
+    const std::vector<int>* relXPositions;
```

Comment:
why not make these static? Seems like they are a property of each piece rather than an ivar.

Reply:
I tried :(